source/Generic/ExtraMetadataToolsVideo/ExtraMetadataToolsVideo.psm1

```diff
@@ -814,98 +1070,58 @@ function Set-YouTubeVideo
-
-        $extraMetadataDirectory = Set-GameDirectory $game
-        $videoPath = Join-Path $extraMetadataDirectory -ChildPath "VideoTrailer.mp4"
-        $videoTempPath = Join-Path $extraMetadataDirectory -ChildPath "VideoTemp.mp4"
-        $youtubedl = $settings.youtubedlPath
-
-        if (Test-Path $videoPath)
+        $success = Download-YouTubeVideo $game $search $false
+        if($success) 
         {
-            continue
-        }
-        if (Test-Path $videoTempPath)
-        {
-            try {
-                Remove-Item $videoTempPath -Force
-            } catch {}
-        }
-        
-        $trailerdownloadparams = @{
-            'FilePath'     = $youtubedl
-            'ArgumentList' = '-o ' + "`"$videoTempPath`"", '-f "mp4"', $search
-            'Wait'         = $true
-            'PassThru'     = $true
-        }
-        $proc = Start-Process @trailerdownloadparams -WindowStyle $settings.youtubedlWindowStyle
-        
-        if (Test-Path $videoTempPath -PathType leaf)
-        {
-            $isConversionNeeded = Get-IsConversionNeeded $videoTempPath
-            if ($isConversionNeeded -eq "invalidFile")
-            {
-                continue
-            }
-            elseif ($isConversionNeeded -eq "true")
-            {
-                $arguments = @("-y", "-i", "`"$videoTempPath`"", "-c:v", "libx264", "-c:a", "mp3", "-vf", "scale=trunc(iw/2)*2:trunc(ih/2)*2", "-pix_fmt", "yuv420p", "`"$videoPath`"")
-                $__logger.Info(("Starting ffmpeg with arguments {0}" -f ($arguments -join ", ")))
-                Start-Process -FilePath $settings.ffmpegPath -ArgumentList $arguments -Wait -WindowStyle Hidden
-            }
-            else
-            {
-                Move-Item $videoTempPath $videoPath -Force
-                $__logger.Info(("Conversion is not needed for video and moved to {0}" -f $videoPath))
-            }
-            try {
-                Remove-Item $videoTempPath -Force
-            } catch {}
-            if (Test-Path $videoPath)
-            {
-                $videoSetCount++
-            }
+            $videoSetCount++;
         }
     }
 
-    # Update assets status of collection
-    Update-CollectionExtraAssetsStatus $gameDatabase $false
-
-    $PlayniteApi.Dialogs.ShowMessage(("Done.`n`nSet video to {0} game(s)" -f $videoSetCount.ToString()), "Extra Metadata Tools")
+    $PlayniteApi.Dialogs.ShowMessage(("Done.`n`nSet video to {0} of {1} selected game(s)" -f $videoSetCount.ToString(), $gameDatabase.Count), "Extra Metadata Tools")
 }
 
 function Get-YoutubeResultsArray
 {
     param (
         [string]$queryInput
     )
+    
+    $settings = Get-Settings
 
-    $query = [uri]::EscapeDataString($queryInput)
-    $uri = "https://www.youtube.com/results?search_query={0}&sp=EgQQARgB" -f $query
-    $webContent = Get-DownloadString $uri
-    $webContent -match 'var ytInitialData = ((.*?(?=(;<\/script>))))' | Out-Null
+    $search = "`"ytsearch20:{0}`"" -f $queryInput
```

Comment:
Just to make sure, since you are passing the raw input to search now. 

Could there be potential issues if you don't escape the string or issues with certain characters?

Reply:
Yeah you're right sorry. This seems not to be so straightforward. I found this (https://gist.github.com/bilbothebaggins/efbbc2312ec72c4967622b340e4b26d6) and it works. But i requires to have that code to be added to the project which isn't too great. An easier solution would be much better. Would you happen to have any ideas? 😅

Reply:
Very rudimentary done in d62f860bcb72053ecc9797252c03ea01790ec09b by removing quotes from search string.
Not perfect but then again it is stable enough for normal use cases and if someone wants to forcefully wreck their query then goodspeed to them. After all were talking a dialogue to download videos in Playnite here and not some vulnerable banking software.